qadence/blocks/primitive.py

```diff
@@ -454,3 +454,34 @@ def _block_title(self) -> str:
 
         s += rf" \[params: {params_str}]"
         return s if self.tag is None else (s + rf" \[tag: {self.tag}]")
+
+
+class ProjectorBlock(PrimitiveBlock):
+    """The abstract ProjectorBlock."""
+
+    name = "ProjectorBlock"
+
+    def __init__(
+        self,
+        ket: str,
+        bra: str,
+        qubit_support: int | tuple[int, ...],
+    ) -> None:
+        if isinstance(qubit_support, int):
+            qubit_support = (qubit_support,)
+        if len(bra) != len(ket):
+            raise ValueError(
+                "Bra and ket must be bitstrings of same length in the 'Projector' definition."
+            )
+        elif len(bra) != len(qubit_support):
```

Comment:
@jpmoutinho I am unsure about that check. Is it correct ? It doesn't feel like it.

Reply:
It looks correct to me, what makes you say that? Since the `bra` is a bitstring corresponding to a basis state it must be of the same length as the number of qubits in the support to which the projector is applied, and the previous check already made sure it is of the same length as `ket`, so everything should be fine.